bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerHttpBridgeHandler.java

```diff
@@ -0,0 +1,74 @@
+package org.openhab.binding.opensprinkler.internal.handler;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.opensprinkler.internal.api.OpenSprinklerApiFactory;
+import org.openhab.binding.opensprinkler.internal.config.OpenSprinklerHttpInterfaceConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OpenSprinklerHttpBridgeHandler extends OpenSprinklerBaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(OpenSprinklerHttpBridgeHandler.class);
+
+    private OpenSprinklerHttpInterfaceConfig openSprinklerConfig = null;
+
+    public OpenSprinklerHttpBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        openSprinklerConfig = getConfig().as(OpenSprinklerHttpInterfaceConfig.class);
+
+        if (openSprinklerConfig == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Could not parse the config for the OpenSprinkler.");
+            return;
+        }
+
+        logger.debug("Initializing OpenSprinkler with config (Hostname: {}, Port: {}, Password: {}, Refresh: {}).",
+                openSprinklerConfig.hostname, openSprinklerConfig.port, openSprinklerConfig.password,
+                openSprinklerConfig.refresh);
+
+        try {
+            openSprinklerDevice = OpenSprinklerApiFactory.getHttpApi(openSprinklerConfig.hostname,
+                    openSprinklerConfig.port, openSprinklerConfig.password);
+        } catch (Exception exp) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Could not create a connection to the OpenSprinkler.");
+            logger.debug("Could not create API connection to the OpenSprinkler device. Exception received: {}",
```

Comment:
No need to call `.toString()`. Usually you do not need to update the ThingStatus AND log. Better add the message of the exception to the thing status. (applies to all ocurrences)

Reply:
Done! :) (btw. all of this code was just taken over from the previous implementation :P)